projects/simple/tests/Tests/Example/Project.hs

```diff
@@ -2,25 +2,44 @@ module Tests.Example.Project where
 
 import Prelude
 
+import Clash.Hedgehog.Sized.Unsigned
 import Test.Tasty
 import Test.Tasty.TH
 import Test.Tasty.Hedgehog
 
-import Hedgehog ((===))
+import qualified Clash.Prelude as C
 import qualified Hedgehog as H
 import qualified Hedgehog.Gen as Gen
 import qualified Hedgehog.Range as Range
 
-import Example.Project (plus)
+-- Import the module containing the `accum` function
+import Example.Project (accum)
 
-prop_plusIsCommutative :: H.Property
-prop_plusIsCommutative = H.property $ do
-  a <- H.forAll (Gen.integral (Range.linear minBound maxBound))
-  b <- H.forAll (Gen.integral (Range.linear minBound maxBound))
-  plus a b === plus b a
+-- Define a Hedgehog property to test the `accum` function
+prop_accum :: H.Property
+prop_accum = H.property $ do
 
-tests :: TestTree
-tests = $(testGroupGenerator)
+  -- Simulate for a random duration between 1 and 100 cycles
+  simDuration <- H.forAll (Gen.integral (Range.linear 1 100))
+
+  -- Generate a list of random unsigned numbers.
+  inp <- H.forAll
+    (Gen.list (Range.singleton simDuration)
+    (genUnsigned Range.linearBounded))
+  let
+    -- Simulate the `accum` function for the pre-existing `System` domain
+    -- and 8 bit unsigned numbers.
+    simOut = C.simulateN simDuration (accum @C.System @8) inp
+
+    -- Calculate the expected output, the result of the last element
+    -- does not appear because the accumulator has 1 cycle latency.
+    expected = init $ scanl (+) 0 inp
```

Comment:
I really dislike `simulate` because it shifts the input and drops the first element of the output, which hinders understanding. It makes this code short and sweet maybe, but as soon as the user starts to think about what the flip-flop is doing in hardware, they think "hey if the reset is asserted in the first cycle why don't I see this with `simulate`". In other words, the moment they understand their design they don't understand this test anymore!

Also, `simulate` is such a drag to use with multiple inputs, which is a really common use case. People will get annoyed and try out `sample` and wonder why it is giving totally different results than `simulate`. I feel `simulate` should be deprecated.

I feel strongly about this, but I'm not going to argue if you let it lie by the road for now and keep the original code.

```suggestion
    -- Simulate the @accum@ function for the pre-existing @System@ domain
    -- and 8 bit unsigned numbers.
    --
    -- The (hidden) reset input of @accum@ will be asserted in the first cycle;
    -- during this cycle it will emit its initial value and the input is
    -- ignored. So we need to present a dummy output value.
    simOut = C.sampleN (simDuration + 1) (accum @C.System @8 (C.fromList (0:inp)))

    -- Calculate the expected output. The first cycle is the initial value, and
    -- the result of the final input value does not appear because the
    -- accumulator has 1 cycle latency.
    expected = 0 : init (scanl (+) 0 inp)
```

Reply:
Ah right, for posterity, as discussed: if somebody feeds a design where some flip-flop does /not/ have a reset to `simulate`, it breaks spectacularly. It's just unintelligible why you get the results you get.

Reply:
IMO we should deprecate `simulate`. It's usefulness isn't clear to me..